### PR TITLE
Developing the developer guide

### PR DESCRIPTION
**Pull request by JuliaKukulies**
This is a draft pull request to collaboratively work on the developer guide.  

I have started to suggest some subsections and points to include based on the discussion in #265 

Here we can look at how our progress looks like on read the docs: https://tobac--281.org.readthedocs.build/en/281/

### CODE REVIEW

**Comment by fsenf** [comment]
@JuliaKukulies : I now added the sphinx-based rendering docu to this PR. I used pandoc to convert markdown to rst, but I am not sure with all code parts have been converted in the correct way. Could you have a look?

**Comment by JuliaKukulies** [comment]
Perfect @fsenf ! I added your files to `index.rst` but for some reason the page does not show up. At a first glance the conversion looks good, but I will check in more detail.

**Comment by fsenf** [comment]
I made some updates to the dev guide which will come with the next direct commit to the branch. 

All looks good beside that these topics still need to be described in detail in `doc/contributing.rst`
- [ ] Github actions / workflow
- [ ] unit testing -> How to add a unit test
- [ ] jupyter notebook examples

It would be great to get support from @freemansw1 and @w-k-jones for describing the first two items. I could care for the last item. 

Should make separate rst file for each topic to make the content easier to diggest?

**Comment by JuliaKukulies** [comment]
Great, thanks for your updates @fsenf ! I think it is a good idea to have separate .rst files for these items as they are a bit more comprehensive. The structure of the dev guide is also just a first draft suggestion, so feel free to change it where necessary! 

What should we do with the CONTRIBUTING.md? Do you want to keep it or should we just make sure that all the information is somewhere in the dev guide? I personally find it better to have all on one place and remove the CONTRIBUTING.md

**Comment by fsenf** [comment]
My suggestion would be: 
- Keep CONTRIBUTING.md 
- but reduce content to nice welcome words and a link list to the actual content in the dev guide

**Comment by JuliaKukulies** [comment]
> My suggestion would be:
> 
> * Keep CONTRIBUTING.md
> * but reduce content to nice welcome words and a link list to the actual content in the dev guide

I like this suggestion- let's do that!

**Comment by JuliaKukulies** [comment]
I converted this from a draft pull request to a real one now and unless @fsenf does not have any oppositions, I think it is finally ready for review.

I have merged the latest changes from the RC branch and included sections on 

- unit testing 
- github actions 
- jupyter notebooks 

I have also made some minor modifications and clean-up to `CONTRIBUTING.md` (see discussion above).

I think the developer guide does not need to be perfect to be merged as we continue working on it. However, a section where I would like particular attention from whoever reviews this PR is [the section on migrating to xarray](https://tobac--281.org.readthedocs.build/en/281/code_structure.html#migrating-to-xarray-and-dask). I have included some informatioon on the decorators, but maybe it would actually be benefial for all of us developers to provide step-by-step instructions on how to work on this topic?

**Comment by fsenf** [comment]
No objections at all! 

@JuliaKukulies: Thank you for finalizing this very important contributions! I will try to make some time next week to contribute to the PR.

File: doc/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+configuration can be found in 
+[.github/workflows](https://github.com/tobac-project/tobac/tree/main/.github/workflows) and encompass
+
+* check that code is formatted using the latest stable version of black 
+* linting of the latest code changes that checks the code quality and results in a score compared to the most recent 
+released version 
+* check of the zenodo JSON file that ensures that the citation is correct 
+* check that all unit tests pass (including testing on multiple operating testings) and report test coverage 
+* check that the example jupyter notebooks run without problems 
+* check that the readthedocs page is properly built
```
Comment (fsenf) [inline doc/contributing.rst:124]: Not sure this is true

File: doc/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+repository](https://zenodo.org/records/10863405). 
+
+=========================
+Releasing a new version 
+=========================
+
+This is the checklist of steps for a release of a new **tobac** version:
+
+* Bump version in `__init__.py `in :code:`hotfix`
+* Add changelog in :code:`hotfix`
```
Comment (fsenf) [inline doc/contributing.rst:161]: not sure if hotfix is the proper naming here. Should it be RC_vX.X.X ?

**Review by fsenf — CHANGES_REQUESTED** [review]
This dev doc is in a very good shape! Thank you a lot for your efforts! I just have a few minor comments and incorporated some spelling and formatting corrections.

File: doc/contributing.rst (hunk 1 above)
Comment (fsenf) [reply doc/contributing.rst:124]: I mean the action on RTD rendering

Comment (JuliaKukulies) [reply doc/contributing.rst]: Good point, I think we are actually not testing this. And i only now saw your detailed description on the sphinx-based rendering and how you can check the RTC page for pull requests etc. So I just removed this point here and think your descriptions suffice.

File: doc/contributing.rst (hunk 2 above)
Comment (JuliaKukulies) [reply doc/contributing.rst]: Thanks for pointing this out. I believe the check list was copy pasted from a hotfix case, but this is confusing so I replaced it with our normal procedure for the  `RC_vXXX` branch

**Comment by JuliaKukulies** [comment]
Thanks for your comments, @fsenf ! I think I addressed them all :)

**Review by fsenf — APPROVED** [review]
Once again, great work!

**Comment by fsenf** [comment]
@freemansw1 : Hi Sean, this PR is ready for a 2nd review. Do you think you can make it until the next dev meeting? Thanks, Fabian.

**Comment by fsenf** [comment]
Who agreed to be the 2nd reviewer here? Was it you @freemansw1 ?

**Review by freemansw1 — APPROVED** [review]

**Comment by freemansw1** [comment]
Thanks, @JuliaKukulies and @fsenf .

We have one minor merge conflict to fix, but I would also be happy for this to be merged directly into `main` as it only touches documentation, per our policies around that kind of thing.

**Comment by JuliaKukulies** [comment]
Not super sure why the zenodo json check fails, since the formatting and syntax seemed to be OK after i updated my affiliation?

**Comment by JuliaKukulies** [comment]
All right, I fixed the json formatting issue, so I am merging directly into `main` as we have discussed before. Thanks @fsenf and @freemansw1 and lets make sure to continue to work and update the developer guide in the future.

**Comment by freemansw1** [comment]
> All right, I fixed the json formatting issue, so I am merging directly into `main` as we have discussed before. Thanks @fsenf and @freemansw1 and lets make sure to continue to work and update the developer guide in the future.

See the JSON formatting CI was making sure that you complied with all new marketing requirements :) (I still don't know what the issue was, but glad it resolved).

**Comment by JuliaKukulies** [comment]
> > All right, I fixed the json formatting issue, so I am merging directly into `main` as we have discussed before. Thanks @fsenf and @freemansw1 and lets make sure to continue to work and update the developer guide in the future.
> 
> See the JSON formatting CI was making sure that you complied with all new marketing requirements :) (I still don't know what the issue was, but glad it resolved).

Haha goood :) 

Also not sure what the issues was, I just copied the json file from another branch and it worked

**Pull request merged**